.github/workflows/ci-graphspace.yaml

```diff
@@ -0,0 +1,32 @@
+name: Docker Build & Push Graphspace
+
+on: [push, pull_request]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: login to docker hub 
+      uses: docker/login-action@v2
+      with: 
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_TOKEN }}
+
+    - name: Set outputs
+      id: vars
+      if: ${{ github.ref == 'refs/heads/master' }}
+      run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
```

Comment:
Put some prefixes that will help in future to distinguish images like gs- or rc- etc